packages/time-series/lib/commands/index.ts

```diff
@@ -168,3 +219,96 @@ export function pushWithLabelsArgument(args: CommandArguments, selectedLabels?:
     return pushVariadicArguments(args, selectedLabels);
   }
 }
+
+export function resp2MapToValue<
+  V extends TuplesReply<[key: BlobStringReply, ...rest: any]>,
+  T
+>(
+  reply: UnwrapReply<ArrayReply<V>>,
+  parseFunc: (v: UnwrapReply<V>) => T,
+  typeMapping?: TypeMapping
+): MapReply<BlobStringReply | string, T> {
+  switch (typeMapping? typeMapping[RESP_TYPES.MAP] : undefined) {
+    case Map: {
+      const ret: Map<BlobStringReply, T> = new Map<BlobStringReply, T>();
+
+      for (let i=0; i < reply.length; i++) {
+        const s = reply[i];
+        const sample = s as unknown as UnwrapReply<typeof s>;
+    
+        ret.set(sample[0], parseFunc(sample));
+      }
+    
+      return ret as unknown as MapReply<string, T>;
+    }
+    case Array: {
+      const ret: Array<BlobStringReply | T> = [];
+
+      for (let i=0; i < reply.length; i++) {
+        const s = reply[i];
+        const sample = s as unknown as UnwrapReply<typeof s>;
+    
+        ret.push(sample[0], parseFunc(sample));
+      }
+    
+      return ret as unknown as MapReply<string, T>;
+    }
+    default: {
+      const ret: Record<string, T> = Object.create(null);
+
+      for (let i=0; i < reply.length; i++) {
+        const s = reply[i];
+        const sample = s as unknown as UnwrapReply<typeof s>;
+        //const key = sample[0] as unknown as UnwrapReply<BlobStringReply>;
+    
+        ret[sample[0].toString()] = parseFunc(sample);
+        //ret[key.toString()] = parseFunc(sample);
+      }
+    
+      return ret as unknown as MapReply<string, T>;
+    }
+  }
+  
+}
+
+export function resp3MapToValue<
```

Comment:
all the transformations can be done in-place, right?

Reply:
conceptually yes.  for simplicity (but not for garbage collector) i recreated the object from scratch, can just have it overwrite the "value".  slightly worried how that would impact iteration, but if its ok to do that,